libs/components/modals/src/lib/modules/modal/modal-adapter.service.ts

```diff
@@ -47,4 +48,50 @@ export class SkyModalAdapterService {
   #removeClassFromBody(className: string): void {
     this.#bodyEl.classList.remove(className);
   }
+
+  public hideHostSiblings(hostElRef: ElementRef): void {
+    const hostElement = hostElRef.nativeElement;
+    const hostSiblings = hostElement.parentElement.children;
+
+    for (let i = 0; i < hostSiblings.length; i++) {
+      const element = hostSiblings[i];
+      if (
+        element !== hostElement &&
+        !element.hasAttribute('aria-live') &&
+        element.nodeName.toLowerCase() !== 'script' &&
+        element.nodeName.toLowerCase() !== 'style'
+      ) {
+        // preserve previous aria-hidden status of elements outside of modal host
+        this.#hostSiblingPreviousValue.set(
+          element,
+          element.getAttribute('aria-hidden')
+        );
+        element.setAttribute('aria-hidden', 'true');
+      }
+    }
+  }
+
+  public unhideOrRestoreHostSiblings(): void {
+    this.#hostSiblingPreviousValue.forEach((previousValue, element) => {
+      // if element had aria-hidden status prior, restore status
+      if (previousValue) {
+        element.setAttribute('aria-hidden', previousValue);
+      } else {
+        element.removeAttribute('aria-hidden');
+      }
+    });
+    this.#hostSiblingPreviousValue.clear();
+  }
+
+  public hidePreviousModal(topModal: Element): void {
```

Comment:
For all four of these methods, the word "hide" is misleading since the modal is not being hidden for sighted users. Even though it's a tad wordy, I would add the qualifier "FromScreenReaders" to the method names so it's 100% clear.
```
public hidePreviousModalFromScreenReaders
```